upgrades/environ.go

```diff
@@ -55,20 +53,24 @@ func (e environConfigGetter) CloudSpec(names.ModelTag) (environs.CloudSpec, erro
 	return e.m.CloudSpec()
 }
 
-func newEnvironUpgradeOperation(op environs.UpgradeOperation) Operation {
+func newEnvironUpgradeOperation(op environs.UpgradeOperation, controllerUUID string) Operation {
 	steps := make([]Step, len(op.Steps))
 	for i, step := range op.Steps {
-		steps[i] = newEnvironUpgradeStep(step)
+		steps[i] = newEnvironUpgradeStep(step, controllerUUID)
 	}
-	return upgradeToVersion{op.TargetVersion, steps}
+	// NOTE(axw) all two of the current environ upgrade steps will happily
+	// run idempotently; there are no post-upgrade steps that will render
+	// them non-runabble. This is here as a backstop, to ensure the upgrades
+	// continue to run until we remove this code.
+	return upgradeToVersion{jujuversion.Current, steps}
 }
 
-func newEnvironUpgradeStep(step environs.UpgradeStep) Step {
+func newEnvironUpgradeStep(step environs.UpgradeStep, controllerUUID string) Step {
 	return &upgradeStep{
 		step.Description(),
 		[]Target{DatabaseMaster},
 		func(Context) error {
-			return step.Run()
+			return step.Run(environs.UpgradeStepParams{controllerUUID})
```

Comment:
Ah - is it more convenient to write the upgrade steps if you don't need to thread this information down to the steps yourself?